src/TestSuite/IntegrationTestCase.php

```diff
@@ -243,6 +259,17 @@ public function enableCsrfToken()
     }
 
     /**
+     * Calling this method will re-store flash messages into the test session
+     * after being removed by the FlashHelper
+     *
+     * @return void
+     */
+    public function enableRememberFlashMessages()
+    {
+        $this->_rememberFlashMessages = true;
```

Comment:
Yes, people could be not checking on the status code but only on the session content.
That could silently not break the test then, but in fact false-positive show still green even though it now does not redirect anymore and consumes the session on some error page instead.